packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -85,10 +90,24 @@ export class GitClientService {
     }
 
     if (pullRequestMode === EnumPullRequestMode.Accumulative) {
-      await this.createBranchIfNotExists({
+      const localRepository = new GitClient();
+      const cloneUrl = `https://${this.provider.domain}/${owner}/${repositoryName}.git`;
+
+      const cloneDir = join(
+        tempDir,
+        this.provider.name,
+        owner,
+        repositoryName,
+        v4()
+      );
+
+      await localRepository.clone(cloneUrl, cloneDir);
```

Comment:
I think it's better to init repository with baseDir, which is closeDir in the current context, then clone.

Reply:
The job of the GitClient is not to hold the context of the clone